students/kegan/session03/mailroom.py

```diff
@@ -0,0 +1,301 @@
+    """ Returns average of given donations rounded to 2 decimals.
+    Args:
+        donations (list of floats) : list of donations
+    Returns:
+        float : average of donations
+    """
+    avg = sum(donations) / len(donations)
+    return round(avg, 2)
+
+
+def dollar(amount, report=False):
+    """ Returns amount in a format suitable for showing
+    dollar amounts, e.g. 181.98 or 211. Removes cents
+    from any whole dollar amount.
+     Args:
+        amount (float) : amount to turn into dollar format
+        report (True) : True returns report format, otherwise basic format
+    Returns:
+        str : amount in dollar format
+    """
+    if report:  # XXXX.XX
+        return '{:.2f}'.format(amount)
+    else:  # X,XXX or X,XXX.XX
+        dollars = get_dollars(amount // 1)
+        cents = get_cents(amount % 1)
+        return dollars + cents
+
+
+def get_dollars(dollars):
+    """ Returns dollar amount as a string with 100s
+    separated by commas (e.g. XX,XXX,XXX)
+    Args:
+        dollars (float) : dollar amount
+    Returns:
+        str : dollars as a string with commas
+    """
+    dollars = str(int(dollars))
+    dollars = ','.join(reversed([
+        dollars[max(0, index - 2):index + 1]
+        for index in list(range(len(dollars)))[::-3]]))
+    return dollars
+
+
+def get_cents(cents):
+    """ Returns cent amount as a string with no
+    leading 0. Returns empty string if cents = 0.
+    Args:
+        cents (float) : cents
+    Returns:
+        str : cents in string format
+    """
+    return '' if cents == 0 else '{:.2f}'.format(cents)[1:]
+
+
+def pad(string, length, trailing):
+    """ Adds leading or trailing whitespace to a string
+    up to a given total length. Orientation gives whether
+    whitespace should be leading or trailing.
+    Args:
+        string (str) : string to pad with whitespace
+        length (int) : desired length of string after padding
+        trailing (str) :
+            True if you want trailing whitespace
+            False if you want leading whitespace
+    Returns:
+        str : string padded with whitespace
+    """
+    while len(string) < length:
+        if trailing:
+            string = string + ' '
+        else:
+            string = ' ' + string
+    return string
+
+
+def stringify(row, separator):
+    """ Returns given row as a string.
+    Args:
+        row (list of str) : list of items in row
+        separator (str) : separator for columns
+    Returns:
+        str : row as a string
+    """
+    line = []
+    trailing = True  # trailing WS on first line
+    for value, width in zip(row, COLUMN_WIDTHS):
+        # monetary values get WS inserted between $ and number
+        if type(value) is float:
+            value = dollar(value, report=True)
+            value = pad(value, width, trailing)
+            value = '$' + value + ' '
+        # numbers have leading WS with one extra WS in front and back
+        elif type(value) is int:
```

Comment:
hmm -- you may not want to switch on type but rather value. or don't switch at all, you should know what type to expect for a given "field".